packages/grid/src/components/Col.tsx

```diff
@@ -1,14 +1,16 @@
 import React, { useMemo } from 'react';
 import cn from 'classnames';
 
+import { PropsBasedOnComponent } from '../../../../shared/types/PropsBasedOnComponent';
 import getGridClassNames from '../utils/getGridClassNames';
 
 import { ColProps } from './Col.types';
 
 import guttersStyles from '../styles/gutters.module.css';
 import styles from './Col.module.css';
 
-function Col<E extends React.ElementType = 'div'>({
+// export Col is necessary with PropsBasedOnComponent due to a Storybook bug
+export const Col: PropsBasedOnComponent<ColProps, 'div'> = ({
```

Comment:
should we keep the default export in this case (IMO we shouldn't) and what is the Storybook bug?